sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveCatalog.scala

```diff
@@ -182,13 +189,15 @@ private[spark] class HiveCatalog(client: HiveClient) extends ExternalCatalog wit
     client.getTable(db, table)
   }
 
+  override def tableExists(db: String, table: String): Boolean = withClient {
+    client.getTableOption(db, table).isDefined
+  }
+
   override def listTables(db: String): Seq[String] = withClient {
-    requireDbExists(db)
```

Comment:
Ah I see. Our current master can allow you call `SHOW TABLES` with a db that does not actually exist and then we displays all temp tables. This behavior can cause confusion to users because they can think that the database exists. But, later when they call `use` to switch to that non-existent db, they will see an exception.